container.go

```diff
@@ -207,6 +207,27 @@ func createContainerWithJSON(id string, c *ContainerConfig, additionalJSON strin
 	return container, nil
 }
 
+// ModifyContainer corresponding to the container id, by sending a request
+func modifyContainer(id string, request *ResourceModificationRequestResponse) error {
```

Comment:
This actually isn't shared, it's just a wrapper used by HNS around the container Modify operation. I still stick with my opinion that this doesn't belong in HCSSHIM at all (it's managing container handle lifetimes) and should be in the caller, but that's not for this PR as it's just being moved here.

I think this should move back to `HNSEndpoint.go`. @jhowardmsft WDYT?